h/api/schemas.py

```diff
@@ -90,9 +205,75 @@ class CreateAnnotationSchema(object):
 
     def __init__(self, request):
         self.request = request
-        self.structure = AnnotationSchema()
+        if self.request.feature('postgres_write'):
+            self.structure = AnnotationSchema()
+        else:
+            self.structure = LegacyAnnotationSchema()
 
     def validate(self, data):
```

Comment:
Please help me understand something here.
By moving the previous implementation of `validate` into `_legacy_validate` and depending on the `postgres_write` feature flag either return the dict that has been sanitized by `validate` (for flag on) or `_legacy_validate` (for flag off), doesn't that mean that we start storing things like `target_uri` instead of the `uri` for the Elasticsearch annotations as soon as we turn the feature flag on?
Since turning the feature flag on means we write to both Elasticsearch and to Postgres. Which means that the JSON structure we return in the API will render `target_uri` instead of `uri` which is what we need.
Or am I missing something here?